stellarphot/photometry/photometry.py

```diff
@@ -255,7 +255,9 @@ def single_image_photometry(ccd_image, sourcelist, camera, observatory_location,
                "SKIPPING THIS IMAGE!")
         return None, None
 
-    # Set high pixels to NaN
+    # Set high pixels to NaN (make sure ccd_image.data is a float array first)
+    if ccd_image.data.dtype != float:
```

Comment:
I think you can just cast it as a `float` without checking the type first. If it is already `float` the cast just doesn't do anything. (Also, use `isinstance` for type checks :))

Reply:
Fixed.